File: app/domain/entity/role.go
package entity

import (
	"time"
)

// Role 角色领域实体
type Role struct {
	ID          uint      `json:"id"`
	Name        string    `json:"name" gorm:"size:100;not null"`
	Code        string    `json:"code" gorm:"uniqueIndex;size:100"`
	NameEn      string    `json:"name_en" gorm:"size:100"`
	Description string    `json:"description" gorm:"size:500"`
	DataScope   string    `json:"data_scope" gorm:"size:50"` // all/own
	Status      int       `json:"status" gorm:"default:1"`   // 1启用 0禁用
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

func (Role) TableName() string {
	return "roles"
}

// IsEnabled 检查角色是否启用
func (r *Role) IsEnabled() bool {
	return r.Status == 1
}

// CreateRoleParams 创建角色参数
type CreateRoleParams struct {
	Name        string
	Code        string
	NameEn      string
	Description string
	DataScope   string
	Status      int
}

// UpdateRoleParams 更新角色参数
type UpdateRoleParams struct {
	Name        *string
	Code        *string
	NameEn      *string
	Description *string
	DataScope   *string
	Status      *int
}

// RoleListQuery 角色列表查询参数
type RoleListQuery struct {
	Keyword string
	Status  *int
	Page    int
	PageSQL int
}

File: frontend/src/api/role.ts
import request from './index'

// ==================== Types ====================

export interface RoleInfo {
  id: number
  name: string
  code: string
  description?: string
  status: number // 1: active, 0: disabled
  created_at: string
  updated_at: string
}

export interface RoleListParams {
  page: number
  page_size: number
  keyword?: string
  status?: number
}

export interface RoleListResponse {
  list: RoleInfo[]
  total: number
  page: number
  page_size: number
}

export interface CreateRoleParams {
  name: string
  code: string
  description?: string
  status?: number
}

export interface UpdateRoleParams {
  name?: string
  code?: string
  description?: string
  status?: number
}

// ==================== Role CRUD APIs ====================

/**
 * 获取角色列表
 */
export function getRoleList(params: RoleListParams) {
  return request.get<RoleListResponse>('/roles', { params })
}

/**
 * 获取角色详情
 */
export function getRole(id: number) {
  return request.get<RoleInfo>(`/roles/${id}`)
}

/**
 * 创建角色
 */
export function createRole(data: CreateRoleParams) {
  return request.post('/roles', data)
}

/**
 * 更新角色信息
 */
export function updateRole(id: number, data: UpdateRoleParams) {
  return request.put(`/roles/${id}`, data)
}

/**
 * 删除角色
 */
export function deleteRole(id: number) {
  return request.delete(`/roles/${id}`)
}

// ==================== Permission APIs ====================

export interface PermissionNode {
  id: number
  name: string
  code: string
  type: 'menu' | 'button'
  parent_id?: number
  children?: PermissionNode[]
}

/**
 * 获取权限树
 */
export function getPermissionTree() {
  return request.get<PermissionNode[]>('/permissions/tree')
}

/**
 * 分配角色权限
 */
export function assignPermissions(roleId: number, permissionIds: number[]) {
  return request.post(`/roles/${roleId}/permissions`, { permission_ids: permissionIds })
}

/**
 * 获取角色拥有的权限
 */
export function getRolePermissions(roleId: number) {
  return request.get<number[]>(`/roles/${roleId}/permissions`)
}
